### PR TITLE
Added Card Object, Unit Card that extends and the Deck object.

### PR DESCRIPTION
**Pull request by smahesh1**
Wallach would not be proud of only making one commit, ya hate to see it

### CODE REVIEW

**Comment by apgiorgio** [comment]
Okay, my thoughts: I feel like location is kinda messy to keep track of as a string.  I think it's okay for now, but once we have Hand and Graveyard objects, I feel like it should be a pointer to some generic CardLocation interface which Hand, Graveyard, Deck, and maybe Board all extend.  

I also think Deck.remove() should take a CardObject as an argument rather than a string; we could also have a function that returns a pointer to a card in the deck with a given name, and in combination those would have the same effect, but it makes the remove function more reusable.

Also, I kinda feel like the name changeAtk implies that the argument will be the new attack value instead of an amount added.  Maybe that's just me though.

File: src/main/kotlin/card/UnitCard.kt [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+   Interface for "Unit" cards - minions that can be played on the board.
+ */
+
+
+    var atk: Int
+    var health: Int
+    var faction: String
+    var cost: Int
+
+    fun changeAtk(amount: Int) {
```
Comment (apgiorgio) [inline src/main/kotlin/card/UnitCard.kt]:
```suggestion
    fun addAtk(amount: Int) {
```

File: src/test/kotlin/card/UnitCardTest.kt [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+            override var text = "The coolest fish in all the land"
+            override var atk = 3
+            override var health = 5
+            override var faction = "Ancients"
+            override var cost = 2
+
+        }
+
+
+        testUnit.changeAtk(5)
```
Comment (apgiorgio) [inline src/test/kotlin/card/UnitCardTest.kt]:
```suggestion
        testUnit.addAtk(5)
```

File: src/test/kotlin/card/UnitCardTest.kt [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        assertEquals(8, testUnit.atk)
+
+        testUnit.changeAtk(-15)
```
Comment (apgiorgio) [inline src/test/kotlin/card/UnitCardTest.kt]:
```suggestion
        testUnit.addAtk(-15)
```

File: src/main/kotlin/deck/Deck.kt [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+     */
+    var cards: MutableList<CardObject> = inpCards
+
+    //Use internal list shuffling
+    fun shuffle(){cards.shuffled()}
+
+
+    //Remove some card from the deck with matching name? id? something we need to figure out
+    //Probably a cleaner way to do this - but couldn't find one
+    fun remove(name: String){
```
Comment (apgiorgio) [inline src/main/kotlin/deck/Deck.kt]:
```suggestion
    fun remove(card: CardObject){
```

File: src/main/kotlin/deck/Deck.kt [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+        cards.remove(cards.filter{ x -> x.name == (name)}[0])
```
Comment (apgiorgio) [inline src/main/kotlin/deck/Deck.kt]:
```suggestion
        cards.remove(cards.filter{ x -> x == card}[0])
```

This probably also needs some error checking, but we can worry about that a bit later.

File: src/test/kotlin/deck/DeckTest.kt [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+        var deck = Deck(deckList)
+
+        //How to test shuffle?
+        deck.shuffle()
+
+
+        assertEquals(2, deck.cards.size)
+
+        deck.remove("Fish Cregly")
```
Comment (apgiorgio) [inline src/test/kotlin/deck/DeckTest.kt]:
```suggestion
        deck.remove(testUnit)
```

File: src/test/kotlin/deck/DeckTest.kt [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
+        deck = Deck(deckList)
+
+        //How to test shuffle?
+        deck.shuffle()
+
+
+        assertEquals(2, deck.cards.size)
+
+        deck.remove("Fish Cregly")
```
Comment (apgiorgio) [inline src/test/kotlin/deck/DeckTest.kt]:
```suggestion
        deck.remove(testUnit)
```

File: src/test/kotlin/card/CardObjectTest.kt [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+import org.junit.Test
+import kotlin.test.assertEquals
+
+class CardObjectTest {
+    @Test
+    fun testLocationChange() {
+        val testUnit = object: CardObject {
+            override var location = "Deck"
+            override var type = "Unit"
+            override var name = "FIsh Cregly"
```
Comment (apgiorgio) [inline src/test/kotlin/card/CardObjectTest.kt]:
```suggestion
            override var name = "Fish Cregly"
```

**Comment by smahesh1** [comment]
Made change for dealing with removing bad input, check that out @apgiorgio

**Comment by smahesh1** [comment]
Otherwise, accepted all changes you suggested

**Pull request merged**